src/block-ref-rewrite.js

```diff
@@ -9,17 +9,24 @@ function createBlockRefRewriteMiddleware (opts = {}) {
     throw Error('BlockRefRewriteMiddleware - mandatory "blockTracker" option is missing.')
   }
 
-  return createAsyncMiddleware(async (req, res, next) => {
+  return createAsyncMiddleware(async (req, _res, next) => {
     const blockRefIndex = blockTagParamIndex(req)
     // skip if method does not include blockRef
-    if (blockRefIndex === undefined) return next()
+    if (blockRefIndex === undefined) {
+      return next()
+    }
     // skip if not "latest"
     let blockRef = req.params[blockRefIndex]
     // omitted blockRef implies "latest"
-    if (blockRef === undefined) blockRef = 'latest'
-    if (blockRef !== 'latest') return next()
+    if (blockRef === undefined) {
+      blockRef = 'latest'
+    }
+    if (blockRef !== 'latest') {
+      return next()
+    }
     // rewrite blockRef to block-tracker's block number
     const latestBlockNumber = await blockTracker.getLatestBlock()
+    // eslint-disable-next-line require-atomic-updates
     req.params[blockRefIndex] = latestBlockNumber
```

Comment:
The error claimed this will cause a race around the condition, but on studying the code it does not seem like a likely condition. Hence I disabled `require-atomic-updates`. Is there any alternate solution?

Reply:
You made the right call here. If we use `json-rpc-engine` correctly, which AFAIK we do, the only code that will modify `req` while this function is executing, is this function.